app/controllers/application_controller.rb

```diff
@@ -39,45 +40,38 @@ def forget(user)
 
   # Logs out the current user.
   def log_out
-  	forget(current_user)
+    forget(current_user)
     session.delete(:user_id)
     @current_user = nil
   end
 
-  # Forgets a persistent session.
-  def forget(user)
-    user.forget
-    cookies.delete(:user_id)
-    cookies.delete(:remember_token)
-  end
+  private
 
-	private
-    # Confirms a logged-in user.
+  # Confirms a logged-in user.
   def logged_in_user
     unless logged_in?
       store_location
-      flash[:danger] = "Please log in."
+      flash[:danger] = 'Please log in.'
       redirect_to login_url
     end
   end
 
   def display_name?
```

Comment:
This method is an overkill really because current_user is accessible from the view and yould just test for that, same one line code.

Reply:
Good day sir , i don't understand the point you are trying to raise for this method, it's true that current_user is accessible from the view but this method operates differently from the current_user method that's why i included it, actually it stores the location which a signed out user wants to reach so as to enable redirection to that location after signing in. So i don't get your point clearly.

Reply:
Hi I was referring to the display_name?method

Reply:
Ok now it's clear thanks for responding.